design/BulkAPI.md

```diff
@@ -309,4 +309,60 @@ kruizeconfigjson: |
 ## Bulk API Threads
 
 - **Control Mechanism:** The number of threads used for bulk API operations can be controlled using the environment
-  variable `bulkThreadPoolSize`.
\ No newline at end of file
+  variable `bulkThreadPoolSize`.
+
+## Experiment Name Format Configuration
+
+- **experimentNameFormat:** The `experimentNameFormat` environment variable is used to define the format for experiment
+  names. For example, if the
+  experiment name should follow the structure:
+
+```
+org_id|source_id|cluster_id|namespace|k8s_object_type|k8s_object_name
+```
+
+then set or define the `experimentNameFormat` as follows:
+
+```
+"experimentNameFormat": "%label:org_id%|%label:source_id%|%label:cluster_id%|%namespace%|%workloadtype%|%workloadname%|%containername%"
+```
+
+When making a /bulk call, ensure the label values used in the experiment name format are passed in the payload's filter
+and include sections, matching the format above.
+
+```json
+{
+  "filter": {
+    "exclude": {
+      "namespace": [],
+      "workload": [],
+      "containers": [],
+      "labels": {
+        "key1": "value1",
+        "key2": "value2"
+      }
+    },
+    "include": {
+      "namespace": [],
+      "workload": [],
+      "containers": [],
+      "labels": {
+        "org_id": "ABCOrga",
+        "source_id": "ZZZ",
+        "cluster_id": "ABG"
```

Comment:
@msvinaykumar Do we always need to specify three labels? Can it be two or just one? I'd suggest that we dont add any mandatory structure around the labels. Maybe we can have a MAX_LABELS defined to 3, but not enforce that it should always be 3.

Reply:
@msvinaykumar - Can you please respond to the above comment.

Reply:
Apologies, I missed checking that. It’s not necessary—this was just an example. The labels can vary in number and are not fixed to org_id; they could be organization_id or any other label."

Reply:
Not enforced to 3 , it can have any number